pkg/internal/cryptoutil/utils.go

```diff
@@ -117,6 +122,67 @@ func lengthPrefix(array []byte) []byte {
 	return arrInfo
 }
 
+// Curve25519KeySize number of bytes in a Curve25519 public or private key
+const Curve25519KeySize = 32
+
+// NonceSize size of a nonce used by Box encryption (Xchacha20Poly1305)
+const NonceSize = 24
+
+// PublicEd25519toCurve25519 takes an Ed25519 public key and provides the corresponding Curve25519 public key
+//  This function wraps PublicKeyToCurve25519 from Adam Langley's ed25519 repo: https://github.com/agl/ed25519
+func PublicEd25519toCurve25519(pub []byte) ([]byte, error) {
+	if len(pub) == 0 {
+		return nil, errors.New("key is nil")
+	}
+	if len(pub) != ed25519.PublicKeySize {
+		return nil, fmt.Errorf("%d-byte key size is invalid", len(pub))
+	}
+	pkOut := new([Curve25519KeySize]byte)
+	pKIn := new([Curve25519KeySize]byte)
+	copy(pKIn[:], pub)
+	success := extra25519.PublicKeyToCurve25519(pkOut, pKIn)
+	if !success {
+		return nil, errors.New("error converting public key")
+	}
+	return pkOut[:], nil
+}
+
+// SecretEd25519toCurve25519 converts a secret key from Ed25519 to curve25519 format
+//  This function wraps PrivateKeyToCurve25519 from Adam Langley's ed25519 repo: https://github.com/agl/ed25519
+func SecretEd25519toCurve25519(priv []byte) ([]byte, error) {
```

Comment:
this function can stay in utils.go as it will be used by jwe encryption too (when we move envelope keys encryption to signature keys)